src/net.c

```diff
@@ -1263,8 +1263,8 @@ coap_read_endpoint(coap_context_t *ctx, coap_endpoint_t *endpoint, coap_tick_t n
   coap_mutex_lock(&e_static_mutex);
 #endif /* COAP_CONSTRAINED_STACK */
 
-  coap_address_init(&packet->src);
-  coap_address_copy(&packet->dst, &endpoint->bind_addr);
+  coap_address_init(&packet->addr_info.remote);
```

Comment:
When `COAP_CONSTRAINED_STACK` is not defined, `packet` points to an uninitialized structure on the stack. As its components are initialized separately without clearing the entire structure first, the following call to `coap_endpoint_get_session(... packet ...)` may fail as previously discussed.

Reply:
Good catch.  addr_info now preset to 0.